src/renderer/src/pass/filters.rs

```diff
@@ -0,0 +1,267 @@
+use gfx;
+use gfx::traits::FactoryExt;
+use gfx::handle::Buffer;
+use gfx::Slice;
+pub use ::target::{ColorFormat, GeometryBuffer};
+
+gfx_defines!(
+    vertex Vertex {
+        pos: [f32; 2] = "a_Pos",
+        tex_coord: [f32; 2] = "a_TexCoord",
+    }
+
+    constant FXAAArg {
+        inverse_texture_size: [f32; 2] = "u_InverseTextureSize",
+    }
+
+    pipeline fxaa {
+        vbuf: gfx::VertexBuffer<Vertex> = (),
+        source: gfx::TextureSampler<[f32; 4]> = "t_Source",
+        out: gfx::RenderTarget<ColorFormat> = "o_Color",
+        inverse_texture_size: gfx::ConstantBuffer<FXAAArg> = "Arg",
+    }
+
+    pipeline null {
+        vbuf: gfx::VertexBuffer<Vertex> = (),
+        source: gfx::TextureSampler<[f32; 4]> = "t_Source",
+        out: gfx::RenderTarget<ColorFormat> = "o_Color",
+    }
+);
+
+pub static FXAA_VERTEX_SRC: &'static [u8] = b"
+    #version 150 core
+
+    in vec2 a_Pos;
+    in vec2 a_TexCoord;
+    out vec2 v_TexCoord;
+
+    void main() {
+        v_TexCoord = a_TexCoord;
+        gl_Position = vec4(a_Pos, 0.0, 1.0);
+    }
+";
+
+pub static FXAA_FRAGMENT_SRC: &'static [u8] = b"
+    #version 150 core
+
+    uniform sampler2D t_Source;
+    layout (std140) uniform Arg {
+        uniform vec2 u_InverseTextureSize;
+    };
+
+    in vec2 v_TexCoord;
+
+    out vec4 o_Color;
+
+    void main() {
+        // o_Color = texture(t_Source, v_TexCoord);
+        float FXAA_SPAN_MAX = 8.0;
+        float FXAA_REDUCE_MIN = 1.0/128.0;
+        float FXAA_REDUCE_MUL = 1.0/8.0;
+
+        vec3 luma = vec3(0.2126, 0.7152, 0.0722);
+        float lumaM = dot(texture(t_Source, v_TexCoord).xyz, luma);
+        float lumaTL = dot(texture(t_Source, v_TexCoord + (vec2(-1.0, -1.0) * u_InverseTextureSize)).xyz, luma);
```

Comment:
I replaced `texture` with `textureOffset` for `lumaTL`, `lumaTR`, `lumaBL`, and `lumaBR`. But since `textureOffset` can only take constant offset `result1` and `result2` can be computed only by using `texture`. So `u_InverseTextureSize` is still required to compute `result1` and `result2`.